tests/solidity/test_close_short.py

```diff
@@ -0,0 +1,364 @@
+"""Close long market trade tests that match those being executed in the solidity repo"""
+import unittest
+
+import elfpy.agents.agent as agent
+import elfpy.markets.hyperdrive.hyperdrive_market as hyperdrive_market
+import elfpy.pricing_models.hyperdrive as hyperdrive_pm
+import elfpy.time as time
+import elfpy.types as types
+
+
+# pylint: disable=too-many-arguments
+# pylint: disable=duplicate-code
+
+
+class TestCloseShort(unittest.TestCase):
+    """
+    Test opening a opening and closing a short in hyperdrive
+    3 failure cases: zero amount, invalid amount, invalid timestamp
+    6 success cases:
+        - close immediately, with regular, and small amounts
+        - redeem at maturity, with zero interest, and with negative interest (skipped)
+        - close halfway thru term, with zero, and negative interest (both skipped)
+    """
+
+    contribution: float = 500_000_000
+    target_apr: float = 0.05
+    term_length: int = 365
+    alice: agent.Agent
+    bob: agent.Agent
+    celine: agent.Agent
+    hyperdrive: hyperdrive_market.Market
+
+    def setUp(self):
+        """Set up agent, pricing model, & market for the subsequent tests.
+        This function is run before each test method.
+        """
+        self.alice = agent.Agent(wallet_address=0, budget=self.contribution)
+        self.bob = agent.Agent(wallet_address=1, budget=self.contribution)
+        block_time = time.BlockTime()
+        pricing_model = hyperdrive_pm.HyperdrivePricingModel()
+        market_state = hyperdrive_market.MarketState(
+            trade_fee_percent=0.0,
+            redemption_fee_percent=0.0,
+        )
+        self.hyperdrive = hyperdrive_market.Market(
+            pricing_model=pricing_model,
+            market_state=market_state,
+            position_duration=time.StretchedTime(
+                days=365, time_stretch=pricing_model.calc_time_stretch(self.target_apr), normalizing_constant=365
+            ),
+            block_time=block_time,
+        )
+        _, wallet_deltas = self.hyperdrive.initialize(self.alice.wallet.address, self.contribution, 0.05)
+        self.alice.wallet.update(wallet_deltas)
+
+    def verify_close_short(
+        self,
+        example_agent: agent.Agent,
+        market_state_before: hyperdrive_market.MarketState,
+        agent_base_paid: float,
+        agent_base_proceeds: float,
+        bond_amount: float,
+        maturity_time: float,
+    ):
+        """Close a short then make sure the market state is correct"""
+        # verify that all of Bob's bonds were burned
+        self.assertFalse(
+            example_agent.wallet.shorts
+        )  # In solidity we check that the balance is zero, but here we delete the entry if it is zero
+        # verify that the bond reserves were updated according to flat+curve
+        # the adjustment should be equal to timeRemaining * bondAmount
+        time_remaining = 0
```

Comment:
Can you locate this time_remaining block to be closer to the test that uses it? (line 82)